tests/virt_autotest/virsh_external_snapshot.pm

```diff
@@ -51,33 +51,35 @@ sub run_test {
     my @vm_hostnames_inactive_array = split(/\n+/, $vm_hostnames_inactive);
 
     foreach my $guest (keys %xen::guests) {
-        record_info "virsh-snapshot", "Creating External Snapshot of guest's disk";
-        my $vm_target_dev     = script_output("virsh domblklist $guest --details | awk '/disk/{ print \$3 }'");
-        my $pre_snapshot_cmd  = "virsh snapshot-create-as $guest";
-        my $diskspec_diskonly = "$vm_target_dev,snapshot=external,file=/var/lib/libvirt/images/$guest.disk-only";
-        $pre_snapshot_cmd = $pre_snapshot_cmd . " --disk-only ";
-        $pre_snapshot_cmd = $pre_snapshot_cmd . " --diskspec " . $diskspec_diskonly;
-        my $ex_snapshot_name = "external-snapshot-$guest";
-        $pre_snapshot_cmd = $pre_snapshot_cmd . " --name " . $ex_snapshot_name;
-        assert_script_run "$pre_snapshot_cmd";
-        assert_script_run "virsh snapshot-list $guest | grep $ex_snapshot_name";
+        if (check_guest_disk_type($guest)) {
```

Comment:
next if not check_guest_disk_type($guest) will need much less code change

Reply:
Had updated code as your mentioned.
Thank you so much for your great help.